config/application-default.properties

```diff
@@ -289,10 +281,143 @@ spring.datasource.driver-class-name=org.h2.Driver
+management.metrics.distribution.percentiles-histogram[timer]=true
+# Timer metrics histogram buckets (database queries, method execution)
+management.metrics.distribution.sla[timer]=0.1ms,0.5ms,1ms,10ms,50ms,100ms,200ms,500ms,1s,2s,5s
+# Enable percentile histograms for HTTP request metrics
+management.metrics.distribution.percentiles-histogram.http.server.requests=true
+# Tag all metrics with service name for filtering
+management.metrics.tags.service_name=${spring.application.name}
+# Tag all metrics with environment for separation
+management.metrics.tags.environment=${spring.profiles.active}
+# Disable Prometheus export (using OTLP instead)
+management.prometheus.metrics.export.enabled=false
+
+### OpenTelemetry Logging Configuration ###
+# Enable log export via OTLP
+management.otlp.logging.export.enabled=true
+# Logs endpoint URL
+management.otlp.logging.endpoint=http://localhost:4318/v1/logs
+# Capture trace/span IDs in logs for correlation
+otel.instrumentation.log4j-appender.experimental.capture-mdc-attributes=trace_id,span_id
+# Enable logging context propagation
+logging.context.enabled=true
+
+### Tracing Configuration ###
+# Production-safe sampling rate (10% of requests)
+management.tracing.sampling.probability=0.1
+# Traces endpoint URL
+management.otlp.tracing.endpoint=http://localhost:4318/v1/traces
+# Record HTTP exchanges for detailed analysis
+management.httpexchanges.recording.enabled=true
+# Enable baggage correlation for distributed traces
+management.tracing.baggage.correlation.enabled=true
+# Enable @Observed annotation support
+management.observations.annotations.enabled=true
+# Enable all built-in Spring Boot observations
+management.observations.enable.all=true
+# Enable runtime telemetry (JVM metrics)
+otel.instrumentation.runtime-telemetry.enabled=true
+# (optional: enable JFR/Java17-specific telemetry)
+otel.instrumentation.runtime-telemetry-java17.enabled=true
+# Enable Spring WebMVC request tracing
+otel.instrumentation.spring-webmvc.enabled=true
+# Enable OpenTelemetry annotation support (@WithSpan)
+otel.instrumentation.opentelemetry-instrumentation-annotations.enabled=true
+# Enable OpenTelemetry annotation support (@WithSpan)
+otel.instrumentation.annotations.enabled=true
+# Parent-based sampling with trace ID ratio
+otel.traces.sampler=parentbased_traceidratio
+# Sampling ratio argument (10%)
+otel.traces.sampler.arg=0.1
+# Enable Spring Boot application lifecycle tracing
+otel.instrumentation.spring-boot.enabled=true
+# Enable Spring Data repository tracing
+otel.instrumentation.spring-data.enabled=true
+# Enable RabbitMQ message tracing
+otel.instrumentation.spring-rabbit.enabled=true
+# Enable Spring Security tracing
+otel.instrumentation.spring-security.enabled=true
+
+### 5) Sensitive Information Configuration (COMMENTED OUT FOR PRODUCTION) ###
+## WARNING: These capture potentially sensitive data - NOT for production
+## Captures HTTP request headers (may contain auth tokens)
```

Comment:
Why should I capture this even in development environments? We have debugging, and loggers would at least not send the information around.